Code/UnitTests/TestFramework/Framework/TestFramework.h

```diff
@@ -241,25 +241,100 @@ class EZ_TEST_DLL ezTestFramework
 
 #if EZ_ENABLED(EZ_PLATFORM_ANDROID)
 #  include <Foundation/Basics/Platform/Android/AndroidUtils.h>
+#  include <TestFramework/Utilities/TestSetup.h>
 #  include <android/log.h>
 #  include <android/native_activity.h>
 #  include <android_native_app_glue.h>
 
+static void ezAndroidHandleCmd(struct android_app* pApp, int32_t cmd);
+int ezAndroidMain(int argc, char** argv);
 
-#  define EZ_TESTFRAMEWORK_ENTRY_POINT_BEGIN(szTestName, szNiceTestName)                                                   \
-    int ezAndroidMain(int argc, char** argv);                                                                              \
-    extern "C" void android_main(struct android_app* app)                                                                  \
-    {                                                                                                                      \
-      ezAndroidUtils::SetAndroidApp(app);                                                                                  \
-      /* TODO: do something with the return value of ezAndroidMain?  */                                                    \
-      /* TODO: can we get somehow get the command line arguments to the android app? Is there even something like that? */ \
-      int iReturnCode = ezAndroidMain(0, nullptr);                                                                         \
-      __android_log_print(ANDROID_LOG_ERROR, "ezEngine", "Test framework exited with return code: '%d'", iReturnCode);     \
-    }                                                                                                                      \
-                                                                                                                           \
-    int ezAndroidMain(int argc, char** argv)                                                                               \
-    {                                                                                                                      \
-      ezTestSetup::InitTestFramework(szTestName, szNiceTestName, 0, nullptr);                                              \
+class ezAndroidTestApplication
```

Comment:
I wasn't aware we have Android stuff directly in TestFramework.h.
Can we move this to a separate file ?

Reply:
Done! I created a `Platform` folder in the TestFramework and extracted the android code and moved the uwp files into it.